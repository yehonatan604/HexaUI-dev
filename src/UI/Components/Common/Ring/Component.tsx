import useTheme from "../../../../Core/Hooks/useTheme";
import { TRing } from "../../../../Data/Types/ComponentTypes/Common/RingBorder/TRingBorder";

const Ring = (props: TRing) => {
  const { children, options, className, ...componentProps } = props;

  // Hooks
  const { colors } = useTheme();

  // Options
  const borerColor = options?.borderVariant || colors.standrad;
  const ringColor = options?.ringVariant || colors.standradDark;
  const bgColor = options?.bgVariant || colors.standradLight;
  const borderWidth = options?.borderThickness || 2;
  const ringWidth = options?.ringThickness || 4;
  const textColor = options?.textVariant || colors.standradDark;
  const padding = options?.padding || 0;
  const rounded = options?.rounded || "md";
  const shadow =
    options?.shadow &&
    `shadow-${options?.shadow?.size || "md"} shadow-${
      options?.shadow?.color || colors.white
    }`;

  // JSX
  return (
    <div
      className={`text-${textColor} p-${padding} rounded-${rounded} border-${borderWidth} border-${borerColor} ring-${ringWidth} ring-${ringColor} bg-${bgColor} ${className} 
      ${shadow}`}
      {...componentProps}
    >
      {children}
    </div>
  );
};

export default Ring;
