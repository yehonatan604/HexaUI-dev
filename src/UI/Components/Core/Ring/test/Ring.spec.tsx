import { render } from "@testing-library/react";
import { test, expect, describe } from "vitest";
import Ring from "../Ring";

describe("Ring Component", () => {
  test("applies custom border variant", () => {
    const { container } = render(
      <Ring options={{ borderVariant: "success" }}>Content</Ring>
    );
    expect(container.firstChild).toHaveClass("border-success");
  });

  test("applies custom ring thickness", () => {
    const { container } = render(<Ring options={{ ringThickness: 4 }}>Content</Ring>);
    expect(container.firstChild).toHaveClass("ring-4");
  });

  test("applies custom padding", () => {
    const { container } = render(<Ring options={{ padding: 4 }}>Content</Ring>);
    expect(container.firstChild).toHaveClass("p-4");
  });

  test("applies custom shadow styles", () => {
    const { container } = render(
      <Ring options={{ shadow: { size: "xl", color: "black" } }}>Content</Ring>
    );
    expect(container.firstChild).toHaveClass("shadow-xl shadow-black");
  });
});
