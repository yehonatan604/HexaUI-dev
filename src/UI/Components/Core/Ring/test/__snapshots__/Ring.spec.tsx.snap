// Vitest Snapshot v1

exports[`Ring Component > renders correctly and matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="text-standard-d p-0 rounded-md border-2 border-standard ring-4 ring-standard-d bg-standard-l undefined 
      undefined"
  >
    Content
  </div>
</DocumentFragment>
`;
